web/src/main/java/com/epam/esm/controller/TagController.java

```diff
@@ -0,0 +1,99 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.dto.TagDTO;
+import com.epam.esm.exception.ValidatorException;
+import com.epam.esm.service.EntityService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.math.BigInteger;
+import java.util.Collection;
+
+/**
+ * Rest controller for tags
+ *
+ * @author Andrey Belik
+ * @version 1.0
+ */
+@RestController
+@RequestMapping("/tags")
+public class TagController {
+  private final EntityService<TagDTO, BigInteger> tagService;
+
+  /**
+   * Constructor
+   *
+   * @param tagService {@link com.epam.esm.service.EntityService}
+   */
+  @Autowired
+  public TagController(EntityService<TagDTO, BigInteger> tagService) {
+    this.tagService = tagService;
+  }
+
+  /**
+   * Find all tags method
+   *
+   * @param certificateId certificate id
+   * @return the response entity
+   */
+  @GetMapping()
+  public ResponseEntity<Collection<TagDTO>> findAll(
+      @RequestParam(value = "certificateId", required = false) BigInteger certificateId) {
+    Collection<TagDTO> tags =
+        certificateId == null ? tagService.findAll() : tagService.findAll(certificateId);
+    return new ResponseEntity<>(tags, HttpStatus.OK);
+  }
+
+  /**
+   * Find tag by id
+   *
+   * @param tagId tag id
+   * @return response entity
+   */
+  @GetMapping("/{id}")
+  public ResponseEntity<?> find(@PathVariable("id") BigInteger tagId) {
+    final TagDTO tag = tagService.find(tagId);
```

Comment:
@belikandrey Could you move all business logic from controller layer to service layer please